sim/core/spell.go

```diff
@@ -142,11 +142,11 @@ type Spell struct {
 	BonusCritRating    float64
 	CastTimeMultiplier float64
 
-	BaseDamageMultiplierAdditive     float64 // Applies an additive multiplier to spell base damage.                          Equivalent to Modifies Spell Effectiveness (8).
-	DamageMultiplier                 float64 // Applies a multiplicative multiplier to full Direct and Periodic spell damage. Equivalent to Mod Damage Done % or similar effects.
-	DamageMultiplierAdditive         float64 // Applies an additive multiplier to full Direct and Periodic spell damage.      Equivalent to Modifies Damage/Healing Done + Modifies Periodic Damage/Healing Done (22).
-	ImpactDamageMultiplierAdditive   float64 // Applies an additive multiplier to just Direct spell damage.                   Equivalent to Modifies Damage/Healing Done.
-	PeriodicDamageMultiplierAdditive float64 // Applies an additive multiplier to just Periodic spell dammage.                Equivalent to Modifies Periodic Damage/Healing Done (22).
+	BaseDamageMultiplierAdditivePct     int64   // Applies an additive multiplier to spell base damage.                          Equivalent to Modifies Spell Effectiveness (8).
+	DamageMultiplier                    float64 // Applies a multiplicative multiplier to full Direct and Periodic spell damage. Equivalent to Mod Damage Done % or similar effects.
+	DamageMultiplierAdditivePct         int64   // Applies an additive multiplier to full Direct and Periodic spell damage.      Equivalent to Modifies Damage/Healing Done + Modifies Periodic Damage/Healing Done (22).
+	ImpactDamageMultiplierAdditivePct   int64   // Applies an additive multiplier to just Direct spell damage.                   Equivalent to Modifies Damage/Healing Done.
+	PeriodicDamageMultiplierAdditivePct int64   // Applies an additive multiplier to just Periodic spell dammage.                Equivalent to Modifies Periodic Damage/Healing Done (22).
```

Comment:
Use ints to represent additive multipliers in order to avoid floating point precision errors when additing/subtracting floats